packages/frontend/src/components/profile/Profile.js

```diff
@@ -231,6 +232,11 @@ export function Profile({ match }) {
 
     const MINIMUM_AVAILABLE_TO_TRANSFER = new BN('10000000000000000000000');
 
+    // show most convenient methods if it was enabled
+    const activeMethodsMap = getActiveMethodsMap(getActiveMethods(userRecoveryMethods));
+    let shouldShowEmail = Boolean(activeMethodsMap.email.publicKey);
+    let shouldShowPhone = DISABLE_PHONE_RECOVERY && Boolean(activeMethodsMap.phone.publicKey);
```

Comment:
All we need to know here is whether the user has recovery methods of either type `email` or `phone`, instead of using helper methods we can just check `userRecoveryMethods` directly:
```js
const shouldShowEmail = userRecoveryMethods.some(({ kind }) => kind === 'email');
const shouldShowPhone = userRecoveryMethods.some(({ kind }) => kind === 'phone');
```

Then the `getActiveMethods*` functions are no longer needed and can be removed. The `publicKey` checks should also not be necessary once we're checking `userRecoveryMethods` directly, since it won't exist if there's not a value for `publicKey`.

Reply:
But then user can't switching off enabled method.
PublicKey check give a possibility to detect if method enabled, then we can show selector and give a chance to disable it.

Reply:
I don't think that's the case, if the method exists in `userRecoveryMethods` then they have it enabled. If they have it enabled then we can show the UI to disable it.

The call to `getActiveMethods` is confusing and unnecessary here because it returns an array with an entry for _all_ recovery methods, regardless of whether the user actually has them. That's why it is required to check for `publicKey` currently, because if it didn't exist in `userRecoveryMethods` then `getActiveMethods` adds it to the array without a value for `publicKey`. So why not check `userRecoveryMethods` directly since it is the actual source of truth? Then the `getActiveMethods*` helper methods copied from `RecoveryContainer` can be removed from this PR.

I've applied the changes above to my local copy to demonstrate the flow in testnet. Note that I start without an email recovery method, can only add one using wallet.testnet.near.org, and once added I can remove it using the code running with my changes:
![2022-07-11 09 55 03](https://user-images.githubusercontent.com/36863574/178317957-e9207828-5e56-4e80-a10b-202060737b9f.gif)